src/odemis/gui/comp/overlay/view.py

```diff
@@ -235,18 +234,19 @@ def on_motion(self, evt):
         if not self.active.value:
             return super(ViewOverlay, self).on_motion(evt)
 
-        if hasattr(self.cnvs, "left_dragging") and self.cnvs.left_dragging:
+        if hasattr(self.cnvs, "left_dragging") and self.cnvs.left_dragging\
+                or hasattr(self.cnvs, "right_dragging") and self.cnvs.right_dragging:
             # Already being handled by the canvas itself
             evt.Skip()
             return
 
-        if self._left_dragging:
+        if self.cnvs.gadget_overlay.on_motion:
+            # the control is passed to the gadget overlay, drawing and editing of tools is possible
             evt.Skip()
 
-        else:
-            vpos = evt.Position
-            self._v_pos = Vec(vpos)
-            self.cnvs.request_drawing_update()
+        vpos = evt.Position
+        self._v_pos = Vec(vpos)
+        self.cnvs.request_drawing_update()
```

Comment:
I think you can simplify a lot by always "skipping" the event.
The only special case which is worthy to deal explicitly with is the dragging, as the position of the image is not directly accessible.
So I'd suggest to simplify the code to this (and it seems to work):
```python
        # Whatever happens, we don't keep the event, and pass it to any other interested listener.
        evt.Skip()

        # If the canvas is being dragged, the image position cannot be directly queried,
        # and anyway the cursor is above the same pixel all the time, so no update.
        if hasattr(self.cnvs, "left_dragging") and self.cnvs.left_dragging:
            return

        self._v_pos = Vec(evt.Position)
        self.cnvs.request_drawing_update()

```